app/models/manageiq/providers/embedded_ansible/automation_manager/configuration_script_source.rb

```diff
@@ -103,4 +113,50 @@ def format_sync_error(error)
     result << error.backtrace.join("\n")
     result.mb_chars.limit(ERROR_MAX_SIZE)
   end
+
+  private
+
+  # Content borrowed from awx
+  #
+  #  https://github.com/ansible/awx/blob/128fa894/awx/main/utils/ansible.py#L17
+  #
+  VALID_PLAYBOOK_CHECK = /^\s*?-?\s*?(?:hosts|include|import_playbook):\s*?.*?$/
+
+  # Confirms two things:
+  #
+  #   - The file extension is a yaml extension
+  #   - The content of the file has one line that matches VALID_PLAYBOOK_CHECK
+  #
+  # Content borrowed from awx and modified from
+  #
+  #   https://github.com/ansible/awx/blob/128fa894/awx/main/utils/ansible.py#L39-L66
+  #
+  def is_a_playbook?(entry, worktree)
+    return false unless entry[:name] =~ /\.ya?ml$/
+
+    worktree.read_entry(entry).lines.each do |line|
+      return true if line =~ VALID_PLAYBOOK_CHECK
+    end
+
+    return false
+  end
+
+  INVALID_DIRS = %w[roles tasks group_vars host_vars]
+
+  # Given a Pathname, remove the following directories from consideration:
+  #
+  #   - roles/
+  #   - tasks/
+  #   - group_vars/
+  #   - host_vars/
+  #
+  # Also ignores hidden files and directories
+  #
+  # Content borrowed from awx and modified from
+  #
+  #   https://github.com/ansible/awx/blob/128fa894/awx/main/utils/ansible.py#L21-L36
+  def not_playbook_dir?(filepath)
+    elements = Pathname.new(filepath).each_filename.to_a
+    elements.any? { |el| el.starts_with?('.') or INVALID_DIRS.include?(el) }
```

Comment:
Was there a reason to use `or` over `||`?

Reply:
I think it looks nice?


¯\\(°\_o)/¯

No, but I saw `rubocop` also disapproves, so I will change.